api/tests.py

```diff
@@ -71,3 +71,83 @@ def test_options_manifest_data(self):
 
+class SpellsTestCase(APITestCase):
+    def setUp(self):
+        from api.management.commands.importer import Importer
+        from api.management.commands.importer import ImportSpec
+        from api.management.commands.importer import ImportOptions
+        from pathlib import Path
+        import json
+        
+        self.test_document_json = """
+            {
+            "title": "Test Reference Document",
+            "slug": "test-doc",
+            "desc": "This is a test document",
+            "license": "Open Gaming License",
+            "author": "John Doe",
+            "organization": "Open5e Test Org",
+            "version": "9.9",
+            "copyright": "",
+            "url": "http://example.com"
+            }
+        """
+        
+        self.test_spell_json = """
+            {
+            "name": "Magic Missile",
+            "desc": "You create three glowing darts of magical force. Each dart hits a creature of your choice that you can see within range. A dart deals 1d4 + 1 force damage to its target. The darts all strike simultaneously, and you can direct them to hit one creature or several.",
+            "higher_level": "When you cast this spell using a spell slot of 2nd level or higher, the spell creates one more dart for each slot level above 1st.",
+            "page": "phb 257",
+            "range": "120 feet",
+            "components": "V, S",
+            "ritual": "no",
+            "duration": "Instantaneous",
+            "concentration": "no",
+            "casting_time": "1 action",
+            "level": "1st-level",
+            "level_int": 1,
+            "school": "Evocation",
+            "class": "Sorcerer, Wizard"
+            }
+        """
+        i = Importer(ImportOptions(update=True, append=False, testrun=False))
+        i.import_document(json.loads(self.test_document_json),ImportSpec("test_filename","test_model_class","import_spell"))
+        i.import_spell(json.loads(self.test_spell_json),ImportSpec("test_filename","test_model_class","import_spell"))
+
+    def test_get_spells(self):
+        response = self.client.get(f'/spells/?format=json')
+        # Confirm the basic elements show up.
+        self.assertContains(response,'count',count=1)
+        self.assertContains(response,'next',count=1)
+        self.assertContains(response,'previous',count=1)
+        self.assertContains(response,'results',count=1)
+
+    def test_get_spell_data(self):
+        import json
+        response = self.client.get(f'/spells/?format=json')
+        # Confirm the basic elements show up.
+        self.assertContains(response,'count',count=1)
+        self.assertContains(response,'next',count=1)
+        self.assertContains(response,'previous',count=1)
+        self.assertContains(response,'results',count=1)
+
+        # Confirm the values appear as expected.
+        self.assertEqual(response.json()['count'],1)
+        in_mm = json.loads(self.test_spell_json)
```

Comment:
What does `mm` mean? I don't think this is semantically meaningful, at least without a clarifying comment.

My best guess is that `mm` could be `monster manual`, but that'd be the wrong source book...